catalog/transformer/catalog-transformer-propertyjson-metacard/src/main/java/ddf/catalog/transformer/metacard/propertyjson/PropertyJsonMetacardTransformer.java

```diff
@@ -0,0 +1,220 @@
+        }
+
+        Map<String, Object> rootObject = new HashMap<>();
+        Map<String, Object> properties = new HashMap<>();
+
+        for (AttributeDescriptor ad : metacard.getMetacardType()
+                .getAttributeDescriptors()) {
+
+            Attribute attribute = metacard.getAttribute(ad.getName());
+            if (attribute != null) {
+                Object value = convertAttribute(attribute, ad, dontInclude);
+                if (value != null) {
+                    properties.put(attribute.getName(), value);
+                }
+            }
+        }
+
+        properties.put(METACARD_TYPE_PROPERTY_KEY,
+                metacard.getMetacardType()
+                        .getName());
+
+        if (StringUtils.isNotBlank(metacard.getSourceId())) {
+            properties.put(SOURCE_ID_PROPERTY, metacard.getSourceId());
+        }
+
+        rootObject.put("properties", properties);
+        return rootObject;
+    }
+
+    @Override
+    public BinaryContent transform(Metacard metacard, Map<String, Serializable> arguments)
+            throws CatalogTransformerException {
+
+        Map<String, Object> rootObject = convertToJSON(metacard);
+
+        String jsonText = json.serialize(rootObject)
+                .toString();
+
+        return new BinaryContentImpl(new ByteArrayInputStream(jsonText.getBytes(StandardCharsets.UTF_8)),
+                DEFAULT_MIME_TYPE);
+    }
+
+    @Override
+    public String toString() {
+        return String.format("%s {Impl=%s, id=%s, MIME Type=%s}",
+                MetacardTransformer.class.getName(),
+                this.getClass()
+                        .getName(),
+                ID,
+                DEFAULT_MIME_TYPE);
+    }
+
+    @Nullable
+    private static Object convertAttribute(Attribute attribute, AttributeDescriptor descriptor,
+            List<AttributeType.AttributeFormat> dontInclude) throws CatalogTransformerException {
+        if (dontInclude.contains(descriptor.getType()
+                .getAttributeFormat())) {
+            return null;
+        }
+
+        if (descriptor.isMultiValued()) {
+            List<Object> values = new ArrayList<>();
+            for (Serializable value : attribute.getValues()) {
+                values.add(convertValue(value,
+                        descriptor.getType()
+                                .getAttributeFormat()));
+            }
+            return values;
+        } else {
+            return convertValue(attribute.getValue(),
+                    descriptor.getType()
+                            .getAttributeFormat());
+        }
+    }
+
+    @Nullable
+    private static Object convertValue(Serializable value, AttributeType.AttributeFormat format)
+            throws CatalogTransformerException {
+        if (value == null) {
+            return null;
+        }
+
+        switch (format) {
+        case DATE:
+            // Creating date format instance each time is inefficient, however
```

Comment:
Have you considered https://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/time/FastDateFormat.html?